kolibri/plugins/coach/assets/src/views/exams-page/change-exam-visibility-modal.vue

```diff
@@ -81,40 +78,48 @@
     },
     data() {
       return {
-        classIsSelected: this.classInitiallySelected(),
-        groupsSelected: this.initiallySelectedGroups(),
+        classIsSelected: this.classIsInitiallySelected(),
+        selectedGroups: this.initiallySelectedGroups(),
       };
     },
-    computed: {
-      groupOptions() {
-        return this.classGroups.map(group => ({
-          label: group.name,
-          id: group.id,
-        }));
-      },
-    },
     methods: {
-      classInitiallySelected() {
+      classIsInitiallySelected() {
         if (this.examVisibility.class) {
           return true;
         }
         return false;
       },
+
       initiallySelectedGroups() {
-        return this.examVisibility.groups.map(group => ({
-          label: group.collection.name,
-          id: group.collection.id,
-        }));
-      },
-      handleSelectChange() {
-        this.classIsSelected = !this.groupsSelected.length;
+        return this.examVisibility.groups.map(group => group.collection.id);
       },
+
       deselectGroups() {
-        this.groupsSelected = [];
+        this.selectedGroups = [];
+      },
+
+      groupIsSelected(groupId) {
+        return this.selectedGroups.includes(groupId);
+      },
+
+      handleGroupChange(groupId, isSelected) {
+        if (isSelected) {
+          if (!this.selectedGroups.includes(groupId)) {
+            this.selectedGroups.push(groupId);
+          }
+        } else {
+          this.selectedGroups = this.selectedGroups.filter(group => group !== groupId);
+        }
+        this.classIsSelected = !this.selectedGroups.length;
+      },
+
+      getGroupName(groupId) {
+        return this.classGroups.find(group => group.id === groupId).name;
```

Comment:
Given that similar code appears several times in this module, it might be worth bringing in `lodash/find`

```diff
--- return this.classGroups.find(group => group.id === groupId).name;
+++ find(this.classGroups, { id: groupId })
```